File: Tutoriels/Java/POO/z0_Ressources/src/manager/GPrint.java
//===============================================
package manager;
//===============================================
public class GPrint {
    //===============================================
    private GPrint() {
        
    }
    //===============================================
    private static class GPrintHolder {       
        private final static GPrint m_instance = new GPrint();
    }
    //===============================================
    public static GPrint Instance() {
        return GPrintHolder.m_instance;
    }
    //===============================================
    public void print(String data) {
        System.out.print(data + "\n");
    }
    //===============================================
}
//===============================================

File: Tutoriels/Cpp/OpenCV/z1_Learning_OpenCV_3/src/manager/GPrint.cpp
//================================================
#include "GPrint.h"
//================================================
GPrint* GPrint::m_instance = 0;
//================================================
GPrint::GPrint() {

}
//================================================
GPrint::~GPrint() {
	
}
//================================================
GPrint* GPrint::Instance() {
	if(m_instance == 0) {
        m_instance = new GPrint;
	}
	return m_instance;
}
//================================================
void GPrint::print(const int& data, const string &name) {
    if(name != "") cout << name << ": ";
    cout << data << "\n";
}
//================================================
void GPrint::print(const double& data, const string &name) {
    if(name != "") cout << name << ": ";
    cout << data << "\n";
}
//================================================
void GPrint::print(const string& data, const string &name) {
    if(name != "") cout << name << ": ";
    cout << data << "\n";
}
//================================================
void GPrint::print(const vector<string>& data, const string &name) {
    if(name != "") cout << name << ": ";
    for(int i = 0; i < data.size(); i++) {
        if(i != 0) cout << " | ";
        string lData = data[i];
        cout << lData;
    }
    cout << "\n";
}
//===============================================
